Release 147.0.0 (#4234)
MetaMask/core: packages/assets-controllers/CHANGELOG.md

## [Unreleased]

## [28.1.0]

### Added

> - Added token detection on 7 more networks ([#4184](https://github.com/MetaMask/core/pull/4184))

Reviewer: Some of these PRs have improved change entries in the PR description, e.g."Token detection is now supported on cronos, celo, gnosis, fantom, polygon_zkevm, moonbeam, and moonriver." here.

The format isn't exactly the same as here though. Maybe we can combine both:

```suggestion
- Addtoken detection on 7 more networks ([#4184](https://github.com/MetaMask/core/pull/4184))
  - New supported networks are: cronos, celo, gnosis, fantom, polygon_zkevm, moonbeam, and moonriver
```

Author: Refactored in [80b8087](https://github.com/MetaMask/core/pull/4234/commits/80b8087d67eec36cd59a5e99c59c4b7a1c3d902c)!

### Changed

- Bump `@metamask/keyring-controller` to ^16.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- Bump `@metamask/keyring-api` to 6.0.0, `@metamask/eth-snap-keyring` to 4.0.0 and snap dependencies ([#4193](https://github.com/MetaMask/core/pull/4193))

### Fixed

> - Fixed NFT detection after closing browser ([#4178](https://github.com/MetaMask/core/pull/4178))

Reviewer: It's unclear to me what is being fixed here as well. I can see that it's related to NFT detection after the browser is closed, but the PR seems to just add initial constructor parameters. It's unclear to me how the additional parameters would be considered a bug fix

Author: Hmm, by looking at the PR description we should probably split this in multiple entries

Reviewer: Yes I think so, good point

> - Fixed update limit ([#4207](https://github.com/MetaMask/core/pull/4207))

Author: I moved this to the Changed section, and I edited the entry to describe the extent of the change

## [Unreleased]

## [29.0.0]

### Added

- Add token detection on 7 more networks ([#4184](https://github.com/MetaMask/core/pull/4184))
  - New supported networks are: cronos, celo, gnosis, fantom, polygon_zkevm, moonbeam, and moonriver

### Changed

- **BREAKING** Changed `NftDetectionController` constructor `options` argument ([#4178](https://github.com/MetaMask/core/pull/4178))
  - Added `options.disabled` and `options.selectedAddress` properties
- **BREAKING** Bump `@metamask/keyring-controller` to ^16.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- **BREAKING** Bump `@metamask/accounts-controller` to ^14.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- **BREAKING** Bump `@metamask/preferences-controller` to ^11.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))

> - Bump `@metamask/keyring-api` to 6.0.0, `@metamask/eth-snap-keyring` to 4.0.0 and snap dependencies ([#4193](https://github.com/MetaMask/core/pull/4193))

Reviewer: Only the one package was bumped in this package:
```suggestion
- Bump `@metamask/keyring-api` to `^6.0.0` ([#4193](https://github.com/MetaMask/core/pull/4193))
```

- Lower number of tokens returned by API calls ([#4207](https://github.com/MetaMask/core/pull/4207))
  - Limit changed from `200` to `50`

> 

Reviewer: Missing dependency bump:

```suggestion
- Bump `@metamask/base-controller` to `^5.0.2` ([#4232](https://github.com/MetaMask/core/pull/4232))

```

## [Unreleased]

## [29.0.0]

### Added

- Add token detection on 7 more networks ([#4184](https://github.com/MetaMask/core/pull/4184))
  - New supported networks are: cronos, celo, gnosis, fantom, polygon_zkevm, moonbeam, and moonriver

### Changed

- **BREAKING** Changed `NftDetectionController` constructor `options` argument ([#4178](https://github.com/MetaMask/core/pull/4178))
  - Added `options.disabled` and `options.selectedAddress` properties
- **BREAKING** Bump `@metamask/keyring-controller` peer dependency to ^16.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- **BREAKING** Bump `@metamask/accounts-controller` peer dependency to ^14.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- **BREAKING** Bump `@metamask/preferences-controller` peer dependency to ^11.0.0 ([#4234](https://github.com/MetaMask/core/pull/4171))
- Bump `@metamask/keyring-api` to `^6.0.0` ([#4193](https://github.com/MetaMask/core/pull/4193))
- Lower number of tokens returned by API calls ([#4207](https://github.com/MetaMask/core/pull/4207))
  - Limit changed from `200` to `50`
- Bump `@metamask/base-controller` to `^5.0.2` ([#4232](https://github.com/MetaMask/core/pull/4232))

> 

Reviewer: Two more missing dependency bumps:

```suggestion
- Bump `@metamask/approval-controller` to `^6.0.2` ([#4234](https://github.com/MetaMask/core/pull/4234))
- Bump `@metamask/polling-controller` to `^6.0.2` ([#4234](https://github.com/MetaMask/core/pull/4234))

```
